xbmc/addons/ScreenSaver.cpp

```diff
@@ -53,27 +45,37 @@ CScreenSaver::~CScreenSaver()
 
   delete m_ifc.screensaver->toAddon;
   delete m_ifc.screensaver->toKodi;
-  delete m_ifc.screensaver->props;
   delete m_ifc.screensaver;
 }
 
 bool CScreenSaver::Start()
 {
-  if (m_ifc.screensaver->toAddon->Start)
-    return m_ifc.screensaver->toAddon->Start(m_ifc.screensaver);
+  if (m_ifc.screensaver->toAddon->start)
+    return m_ifc.screensaver->toAddon->start(m_ifc.hdl);
   return false;
 }
 
 void CScreenSaver::Stop()
 {
-  if (m_ifc.screensaver->toAddon->Stop)
-    m_ifc.screensaver->toAddon->Stop(m_ifc.screensaver);
+  if (m_ifc.screensaver->toAddon->stop)
+    m_ifc.screensaver->toAddon->stop(m_ifc.hdl);
 }
 
 void CScreenSaver::Render()
 {
-  if (m_ifc.screensaver->toAddon->Render)
-    m_ifc.screensaver->toAddon->Render(m_ifc.screensaver);
+  if (m_ifc.screensaver->toAddon->render)
+    m_ifc.screensaver->toAddon->render(m_ifc.hdl);
 }
 
-} /* namespace ADDON */
+void CScreenSaver::GetProperties(struct KODI_ADDON_SCREENSAVER_PROPS* props)
+{
+  if (!props)
+    return;
+
+  props->x = 0;
+  props->y = 0;
+  props->device = CServiceBroker::GetWinSystem()->GetHWContext();
+  props->width = CServiceBroker::GetWinSystem()->GetGfxContext().GetWidth();
+  props->height = CServiceBroker::GetWinSystem()->GetGfxContext().GetHeight();
+  props->pixelRatio = CServiceBroker::GetWinSystem()->GetGfxContext().GetResInfo().fPixelRatio;
```

Comment:
Only call `CServiceBroker::GetWinSystem()` once, assign to a variable and reuse that var.

Reply:
Thanks for the comments, also added now a pointer check to it (was also requested for other places by my font cleanup request).